cf/data/dask_utils.py

```diff
@@ -184,6 +186,130 @@ def cf_harden_mask(a):
     return a
 
 
+def cf_percentile(a, q, axis, interpolation, keepdims=False, mtol=1):
+    """Compute percentiles of the data along the specified axes.
+
+    See `cf.Data.percentile` for further details.
+
+    .. note:: This function correctly sets the mask hardness of the
+              output array.
+
+    .. versionadded:: TODODASK
+
+    .. seealso:: `cf.Data.percentile`
+
+    :Parameters:
+
+        a: `numpy.ndarray`
+            Input array,
+
+        q: `numpy.ndarray`
+            Percentile or sequence of percentiles to compute, which
+            must be between 0 and 100 inclusive.
+
+        axis: `tuple` of `int`
+            Axes along which the percentiles are computed.
+
+        interpolation: `str`
+            Specifies the interpolation method to use when the desired
+            percentile lies between two data points ``i < j``. Must be
+            one of ``'linear'``, ``'lower'``, ``'higher'``,
+            ``'midpoint'``, or ``'nearest'``.
+
+        keepdims: `bool`, optional
+            If this is set to True, the axes which are reduced are
+            left in the result as dimensions with size one. With this
+            option, the result will broadcast correctly against the
+            original array *a*.
+
+        mtol: number, optional
+            Set an upper limit of the amount input data values which
+            are allowed to be missing data when contributing to
+            individual output percentile values. It is defined as a
+            fraction (between 0 and 1 inclusive) of the contributing
+            input data values. The default is 1, meaning that a
+            missing datum in the output array only occurs when all of
+            its contributing input array elements are missing data. A
+            value of 0 means that a missing datum in the output array
+            occurs whenever any of its contributing input array
+            elements are missing data.
+
+    :Returns:
+
+        `numpy.ndarray`
+
+    """
+    if not len(a):
```

Comment:
I guess there are some special cases you want to cater for that aren't covered by:

```suggestion
    if not a:
```
? Just to check in case we can simplify to this...

Reply:
Good to check. This test is for `a` not being scalar, rather than `bool(a)` being False, which would be the case for `np.array(0)`. I'll put a comment in.

This case is not tested, I realise (why return `None`?). So I'll do some further investigations

Reply:
... well that was a can of worms! I think https://github.com/NCAS-CMS/cf-python/pull/313/commits/48cd3fbce272647e1965a56a758436d0a1151383 sorts it all out, though.

Reply:
> Good to check. This test is for a not being scalar, rather than bool(a) being False, which would be the case for np.array(0). I'll put a comment in.

Noted for future Dask work (and otherwise). Thanks for clarifying and adding a comment too which is always useful.